miasm2/expression/expression_dissector.py

```diff
@@ -0,0 +1,168 @@
+import miasm2.expression.expression as m2_expr
+
+
+class ExprDissector:
+    """
+    Dissects expressions into subexpressions
+    """
+
+    def __init__(self, registers=None, irdst=None):
+        """
+        Initialises the architecture variables
+        :param registers: registers of an architecture
+        :param irdst: IRA instruction pointer of an architecture
+        """
+        # set architecture variables
+        if registers and irdst:
+            self.arch_vars = set(registers + [irdst])
```

Comment:
I'm not a big fan of attributes which are sometimes defined, sometimes not.
You can always defined it, with `set()` when no registers are provided.

Then, instead of raising an `AssertionError`, you can raise an error more specific (or a silent pass).